### PR TITLE
Cleanup .NET project files

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Cleanup some .NET project files. Remove deprecated properties and ones that are already defaulted to a correct value. Remove mono handling.

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "projects": [ "src" ]
+  "projects": [ "src" ],
+  "sdk": {
+    "version": "6.0.200",
```
Comment (pakrym-stripe) [inline global.json:4]: We are building for .NET 6, this is the de-facto required SDK version.

File: src/Stripe.net.sln [hunk 2]
```diff
@@ -23,25 +23,17 @@ EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
-		Debug|x64 = Debug|x64
```
Comment (pakrym-stripe) [inline src/Stripe.net.sln:26]: Don't need per-cpu configurations, our libraries are not runtime-specific.

File: src/Stripe.net/Stripe.net.csproj [hunk 3]
```diff
@@ -1,23 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="..\..\netfx.props" />
 
   <PropertyGroup>
     <Description>Stripe.net is a sync/async .NET 4.6.1+ client, and a portable class library for the Stripe API.  (Official Library)</Description>
-    <AssemblyTitle>Stripe.net</AssemblyTitle>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:6]: Defaults to project name.

File: src/Stripe.net/Stripe.net.csproj [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
     <Version>39.107.0</Version>
-    <VersionPrefix>$(Version)</VersionPrefix>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:8]: Version covers this.

File: src/Stripe.net/Stripe.net.csproj [hunk 5]
```diff
@@ -1,23 +1,16 @@
...
     <LangVersion>8</LangVersion>
     <Authors>Stripe, Jayme Davis</Authors>
     <TargetFrameworks>net5.0;net6.0;netcoreapp3.1;netstandard2.0;net461</TargetFrameworks>
-    <AssemblyName>Stripe.net</AssemblyName>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:12]: Defaults to project name.

File: src/Stripe.net/Stripe.net.csproj [hunk 6]
```diff
@@ -1,23 +1,16 @@
...
-    <PackageId>Stripe.net</PackageId>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:13]: Defaults to project name.

File: src/Stripe.net/Stripe.net.csproj [hunk 7]
```diff
@@ -1,23 +1,16 @@
...
     <PackageTags>stripe;payment;credit;cards;money;gateway;paypal;braintree</PackageTags>
-    <PackageIconUrl>http://i.imgur.com/UuBwQ33.png</PackageIconUrl>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:15]: Deprecated.

File: src/Stripe.net/Stripe.net.csproj [hunk 8]
```diff
@@ -1,23 +1,16 @@
...
     <PackageIcon>icon.png</PackageIcon>
     <PackageProjectUrl>https://github.com/stripe/stripe-dotnet</PackageProjectUrl>
     <PackageLicenseFile>LICENSE</PackageLicenseFile>
     <PublishRepositoryUrl>true</PublishRepositoryUrl>
-    <RuntimeIdentifiers>win10-x64</RuntimeIdentifiers>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:20]: We don't have runtime-specific assets.

File: src/Stripe.net/Stripe.net.csproj [hunk 9]
```diff
@@ -39,6 +32,10 @@
     <CodeAnalysisRuleSet>..\_stylecop\StyleCopRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" PrivateAssets="All" />
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:36]: This covers building for .NET framework on non-windows.

File: src/Stripe.net/Stripe.net.csproj [hunk 10]
```diff
@@ -1,23 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="..\..\netfx.props" />
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:2]: Coverted by `Microsoft.NETFramework.ReferenceAssemblies` below.

**Review by kamil-stripe — APPROVED** [review]

**Comment by pakrym-stripe** [comment]
I also confirmed that resulting package output is equivalent.

**Pull request merged**